OpenSim/Common/SmoothSegmentedFunction.cpp

```diff
@@ -39,6 +44,324 @@ static double UTOL = (double)SimTK::Eps*1e2;
 static double INTTOL = (double)SimTK::Eps*1e2;
 static int MAXITER = 20;
 static int NUM_SAMPLE_PTS = 100;
+
+//=============================================================================
+// PARAMETERS
+//=============================================================================
+
+// Helper struct containing all parameters required to construct the
+// SmoothSegmentedFunctionData.
+namespace {
+struct SmoothSegmentedFunctionParameters
+{
+
+    SmoothSegmentedFunctionParameters(
+        const SimTK::Matrix& mX,
+        const SimTK::Matrix& mY,
+        double x0,
+        double x1,
+        double y0,
+        double y1,
+        double dydx0,
+        double dydx1,
+        bool computeIntegral,
+        bool intx0x1):
```

Comment:
Conventionally, there's a space separator between the constructor's parameter list and the initializer list (e.g. `Ctor(T arg, T arg2) : val_(arg), val2_(arg2) {}`)

(unsure why, probably initially related to your form resembling a `goto` label if you squint)